source/inputCore.py

```diff
@@ -436,6 +443,9 @@ def executeGesture(self, gesture):
 		elif speechEffect in (gesture.SPEECHEFFECT_PAUSE, gesture.SPEECHEFFECT_RESUME):
 			queueHandler.queueFunction(queueHandler.eventQueue, speech.pauseSpeech, speechEffect == gesture.SPEECHEFFECT_PAUSE)
 
+		if gesture.shouldPreventSystemIdle:
+			winKernel.SetThreadExecutionState(winKernel.ES_SYSTEM_REQUIRED)

```

Comment:
I do think that we should also include ES_DISPLAY_REQUIRED here, as this will cause Windows not to power off the display. See:
https://docs.microsoft.com/en-us/windows/win32/power/system-sleep-criteria
I do agree it is confusing whether or not it will specifically disable the screen saver though due to the comment in the documentation for SetThreadExecutionState.

Reply:
Makes sense. I initially thought that it would actually be an energy savings benefit that the screen would go black if you have a short screen idle timeout, but it is probably confusing that it stops one idle timer and not the other.